python/sqlalchemy/security/sqlalchemy-execute-raw-query.py

```diff
@@ -193,3 +193,34 @@
 engine = create_engine('postgresql://user@localhost/database')
 query = select(literal_column("users.fullname", String) + ', ' + literal_column("addresses.email_address").label("title")).where(and_(literal_column("users.id") == literal_column("addresses.user_id"), text("users.name BETWEEN 'm' AND 'z'"), text("(addresses.email_address LIKE :x OR addresses.email_address LIKE :y)"))).select_from(table('users')).select_from(table('addresses'))
 conn.execute(query, {"x":"%@aol.com", "y":"%@msn.com"}).fetchall()
+
+
+# SQL Composition using SQL Expression
+# ok: sqlalchemy-execute-raw-query
```

Comment:
Unless I'm missing something, I think you want these `ok` annotations right above the `execute` line :+1:

Reply:
I have to check if concatenation is use before execution:

`# Insert multi data record using SQL Expression
connection_string = 'sqlite:///db.sqlite'
engine = create_engine(connection_string, echo=True)
with engine.connect() as connection:
  meta = MetaData()
  meta.reflect(bind=connection)
  product_table = meta.tables['product']
  # ruleid: sqlalchemy-execute-raw-query
  stmt = insert(product_table) + 'test'
  values = [
      {field_name: 'hazelnut', field_price: 5},
      {field_name: 'banana', field_price: 8}
  ]
  connection.execute(stmt, values)`

`ok` annotations should be just before `stmt` line ?